### PR TITLE
Adds retention policy support

### PR DESCRIPTION
**Pull request by amandakarina**
Hey folks,

This pull request adds support for retention policy in bucket module.

This change is required for the issue https://github.com/terraform-google-modules/terraform-example-foundation/issues/256

Could you please take a look on it?

Thanks!

### CODE REVIEW

**Comment by umairidris** [comment]
Can we implement it as retention_period_days to match expiration_days (https://github.com/terraform-google-modules/terraform-google-log-export/commit/add774ac1b7274a42a8c9bab8e466e53a64169aa#diff-385d394556509a18614d0a5775729fabe0569fe4305712a8c7177ecad2c77060)

**Comment by morgante** [comment]
+1, please include units where possible.

**Comment by umairidris** [comment]
(just to be clear, in addition to including units for documentation, having a bucket lock in seconds which is the default is not useful for audit logs as every org is usually doing it in periods of months or years, so days as a unit is a good middle ground)

File: modules/storage/variables.tf [hunk 1]
```diff
@@ -52,3 +52,12 @@ variable "expiration_days" {
   type        = number
   default     = null
 }
+
+variable "retention_days" {
```
Comment (umairidris) [inline modules/storage/variables.tf:56]:
```suggestion
variable "retention_policy" {
```

File: modules/storage/variables.tf [hunk 2]
```diff
@@ -52,3 +52,12 @@ variable "expiration_days" {
...
+  description = "Configuration of the bucket's data retention policy for how long objects in the bucket should be retained."
+  type = object({
+    is_locked        = bool
+    retention_period = number
```
Comment (umairidris) [inline modules/storage/variables.tf:60]:
```suggestion
    retention_period_days = number
```

File: modules/storage/main.tf [hunk 3]
```diff
@@ -52,6 +52,14 @@ resource "google_storage_bucket" "bucket" {
       }
     }
   }
+
+  dynamic "retention_policy" {
+    for_each = var.retention_days == null ? [] : [var.retention_days]
+    content {
+      is_locked        = var.retention_days.is_locked
+      retention_period = var.retention_days.retention_period
```
Comment (umairidris) [inline modules/storage/main.tf:60]:
```suggestion
      retention_period = var.retention_policy.retention_period_days * 24 * 60 * 60 // days to seconds
```

**Review by umairidris — CHANGES_REQUESTED** [review]

**Review by umairidris — APPROVED** [review]

**Pull request merged**